src/sql/workbench/services/connection/browser/connectionController.ts

```diff
@@ -61,39 +63,35 @@ export class ConnectionController implements IConnectionComponentController {
 		tempProfile.groupFullName = '';
 		tempProfile.saveProfile = false;
 		let uri = this._connectionManagementService.getConnectionUri(tempProfile);
-		return new Promise<string[]>((resolve, reject) => {
-			if (this._databaseCache.has(uri)) {
-				let cachedDatabases: string[] = this._databaseCache.get(uri);
-				if (cachedDatabases !== null) {
-					resolve(cachedDatabases);
+		if (this._databaseCache.has(uri)) {
+			let cachedDatabases: string[] = this._databaseCache.get(uri);
+			if (cachedDatabases !== null) {
+				return cachedDatabases;
+			} else {
+				throw new Error();
```

Comment:
Probably want some sort of error message.

Reply:
There was technically no error message before. I can add something.

Reply:
There was technically no error message before. I can add something.

Reply:
True, but having even just a vague string is useful for debugging when we do get errors and following an error back to a code block.